src/NodeHelper.groovy

```diff
@@ -0,0 +1,1136 @@
+        if (computer != null) {
+            ret = computer.getDescription();
+        }
+
+        return ret;
+    }
+
+    /**
+     * Gets cpu count via exec on the computer passed
+     * in.
+     *
+     * @param compterName computer for which CPU count is needed
+     *
+     * @return number of CPUs on the machine as int.
+     *         -1 if computer not found and 
+     *         -2 if command was unsuccessful
+     */
+    public String getCpuCount(String computerName) {
+        String ret = "getCpuCout: COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            switch (getOsKernelInfo(computer.getName()).get(0)) {
+                case "linux":
+                    ret = getCpuCountFromLinux(computer);
+                    break;
+                case "windows":
+                    ret = getCpuCountFromWindows(computer);
+                    break;
+                case "aix":
+                    ret = getCpuCountFromAix(computer);
+                    break;
+                case "zos":
+                    // ret =
+                    break; 
+                case "mac":
+                    ret = getCpuCountFromMac(computer);
+                    break;
+                default:
+                    ret = "getCpuCout: INVALID_OS_KERNEL/NOT_IMPLEMENTED_YET";
+                    break;
+            }
+        }
+
+        return ret;
+    }
+
+    private String getCpuCountFromLinux(Computer computer) {
+        String ret = "getCpuCountFromLinux: COMPUTER_NOT_FOUND";
+
+        if (computer != null) {
+            /* Probably should add more error checking
+             * if is used somewhere else besides getCpuCount
+             */
+            ret = "getCpuCountFromLinux: UNSUCCESSFUL_EXECUTION";
+            String cmdResult = execGroovy('lscpu', computer);
+
+            if (cmdResult.length() > 0 && !cmdResult.equals("error")) {
+                String[] cmdResultArray = cmdResult.split("\\n");
+
+                int index = 0;
+                while (index < cmdResultArray.length && !cmdResultArray[index].matches("^CPU\\(s\\):\\s{2,}.*")) {
+                    index++;
+                }
+                ret = (cmdResultArray[index].split("^CPU\\(s\\):\\s{2,}"))[1];
+            }
+        }
+
+        return ret;
+    }
+
+    private String getCpuCountFromWindows(Computer computer) {
+        String ret = "getCpuCountFromWindows: COMPUTER_NOT_FOUND";
+
+        if (computer != null) {
+            ret = -2;
+            
+            /* Below is the code with the right way to get CPU count
+             * But due to a bug in setting up the environment variables on windows
+             * this returns an inaccurate result.
+             * This has to be fixed by the person who did the setup for the virtual
+             * machines.
+             * As of right now, we do not know how to fix the bug.
```

Comment:
TODO this

Reply:
Done! [Bug with cpu count on windows machine](https://github.com/AdoptOpenJDK/openjdk-jenkins-helper/issues/6)

Reply:
This was actually bad on my part. I haven't had a chance to check this bug on external machines. I'll run tests regarding this on the external machines and open an issue if needed.